### DISCUSSION TITLE
What am I missing?

### DISCUSSION THREAD

**Discussion by icsy7867** [ipxe/ipxe#825] [General]
I am trying to learn. I have been through the documentation and I have been looking through examples, but there is something I am missing.  Currently I am just trying to do a simple network boot via http using some well known linux mirrors.  My current attempt is rocky linux.

I have been playing around with the kernel parameters, but I can't seem to figure out anything that will work.  I borrowed some of this from the netboot.xyz entry.

```
#!ipxe
dhcp
set ipparam ip=${ip}::${gateway}:${netmask}:::none nameserver=${dns}
set os_arch x86_64

set base http://download.rockylinux.org/pub/rocky/9/BaseOS/x86_64/os/images/pxeboot
set repo http://download.rockylinux.org/pub/rocky/9/BaseOS/x86_64/os

set params inst.text

imgfree
kernel ${base}/vmlinuz inst.repo=${repo} ${params} ${ipparam} initrd=initrd
initrd ${base}/initrd.img
boot
```

When I boot to the image, the two files download just fine and then try to install.  But I am getting various dracut-initqueue and other errors.  The current one is:
```
FATAL: Failed to find a root filesystem in /tmp/curl_fetch_url1/install.img
```

I am having trouble getting a working http example.  100% sure this is user error, I just dont know what else to try.  I have searched the old forums, the dicussions and google.

**Comment by NiKiZe** [comment]
The error you are seeing is distribution specific, probably the repo part isn't working, but again that is distribution specific.

A working example can be seen at boot.ipxe.org/demo

**Comment by icsy7867** [comment]
I think what I am hearing is "ipxe is working as intended.  You need to get it working for this specific distro".

Are there any suggestions on how to read/learn what to use for the kernel parameters for various OS's?

Also, I got this to work.  I used the KPs I found here:
https://github.com/hkkitlee/Ryanboot-offical-linux-netbooter/blob/main/oln.ipxe

I also bumped the RAM. I think some/part/all of my issues were I didnt give the system enough memory as well.

**Reply by NiKiZe** [reply]
The kernel is working just fine, it runs, finds the initrd that was provided and then starts to execute the scripts provided in the initrd. The scripts often parses the cmdline and does things. But each distro implements these scripts on their own, using different parameters or different ways.

I have made this easy for the Gentoo minimal, which you can run from http://gentoo.ipxe.se/boot.ipxe
Try it from iPXE shell as

    chain http://gentoo.ipxe.se/boot.ipxe

**Reply by icsy7867** [reply]
Thank you!  The examples are super helpful and I am starting to get a better grasp on these things.
